nemo/collections/nlp/models/duplex_text_normalization/duplex_tagger.py

```diff
@@ -158,20 +158,24 @@ def _infer(self, sents: List[List[str]], inst_directions: List[str]):
         logits = self.model(**encodings.to(self.device)).logits
         pred_indexes = torch.argmax(logits, dim=-1).tolist()
 
-        # Extract all_tag_preds
+        # Extract all_tag_preds for words
         all_tag_preds = []
         batch_size, max_len = encodings['input_ids'].size()
         for ix in range(batch_size):
-            raw_tag_preds = [constants.ALL_TAG_LABELS[p] for p in pred_indexes[ix][1:]]
+            raw_tag_preds = [
+                constants.ALL_TAG_LABELS[p] for p in pred_indexes[ix][2:]
+            ]  # remove first special token and task prefix token
             tag_preds, previous_word_idx = [], None
             word_ids = encodings.word_ids(batch_index=ix)
```

Comment:
Just set word_ids = encodings.word_ids(batch_index=ix)[2:] so you won't need the 3 lines from 171 to 173?

Reply:
thanks will update